File: hosts/rust/examples/basic_host.rs
use std::collections::BTreeMap;
use std::env;
use std::path::PathBuf;
use evoker_host::PluginClient;
use xmlrpc::Value;

fn main() {
    let cwd = env::current_dir().unwrap();
    let plugins_dir = cwd.join("..").join("test_assets").join("plugins");
    
    let src_dir = cwd.join("..").join("python").join("src");
    unsafe {
        env::set_var("PYTHONPATH", src_dir);
    }
    let worker_script = "plugin_host.worker";

    let mut client = PluginClient::new(&plugins_dir, None, None);
    
    println!("Starting worker...");
    if let Err(e) = client.start_worker(Some("python"), worker_script) {
        eprintln!("Failed to start worker: {}", e);
        return;
    }
    
    println!("Worker started! Scanning plugins...");
    match client.scan() {
        Ok(manifest) => println!("Plugins Manifest: {:?}", manifest),
        Err(e) => eprintln!("Scan error: {}", e),
    }

    println!("Invoking test_plugin.hello_world...");
    let mut kwargs = BTreeMap::new();
    kwargs.insert("name".to_string(), Value::String("Rust Developer".to_string()));
    
    match client.invoke("test_plugin", "hello_world", kwargs) {
        Ok(res) => println!("Result: {:?}", res),
        Err(e) => eprintln!("Invoke error: {}", e),
    }
}

File: hosts/cpp/examples/basic_host.cpp
#include <iostream>
#include <filesystem>
#include "evoker_client.hpp"

int main() {
    std::filesystem::path current = std::filesystem::current_path();
    std::filesystem::path plugins_dir = current / ".." / ".." / "test_assets" / "plugins";
    std::filesystem::path src_dir = current / ".." / ".." / "python" / "src";
    std::string worker_script = "evoker.worker";

    if (!std::filesystem::exists(plugins_dir)) {
        plugins_dir = current / ".." / "test_assets" / "plugins";
        src_dir = current / ".." / "python" / "src";
    }

#ifdef _WIN32
    _putenv_s("PYTHONPATH", src_dir.string().c_str());
#else
    setenv("PYTHONPATH", src_dir.string().c_str(), 1);
#endif

    evoker::PluginClient client(plugins_dir.string());
    
    std::cout << "Starting worker..." << std::endl;
    if (!client.start_worker("python", worker_script)) {
        std::cerr << "Failed to start worker" << std::endl;
        return 1;
    }

    std::cout << "Scanning plugins..." << std::endl;
    auto manifest = client.scan();
    std::cout << "Manifest: " << manifest.dump(2) << std::endl;

    std::cout << "Invoking test_plugin..." << std::endl;
    auto result = client.invoke("test_plugin", "hello_world", {{"name", "World"}});
    std::cout << "Result: " << result.dump() << std::endl;

    return 0;
}
